### PR TITLE
Add list views for workspaces and requests

### PR DESCRIPTION
**Pull request by bloodearnest**
- Add list pages for Workspaces and Requests
- Basic navigation in header to list pages.
- Remove support for navigating from WorkspaceRoot object

### CODE REVIEW

File: tests/integration/test_views.py [hunk 1]
```diff
@@ -209,3 +197,22 @@ def test_request_view_redirects_to_file(client_with_permission, tmp_request):
         response.headers["Location"]
         == f"/requests/{tmp_request.workspace}/{tmp_request.request_id}/file.txt"
     )
+
+
+def test_requests_index_user_permitted_requests(client_with_user):
+    WorkspaceFactory("test1").create_request("test-request")
+    permitted_client = client_with_user({"id": 1, "workspaces": ["test1"]})
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply tests/integration/test_views.py:204]: Ah, nice, will do

File: tests/integration/test_views.py [hunk 2]
```diff
@@ -209,3 +197,22 @@ def test_request_view_redirects_to_file(client_with_permission, tmp_request):
...
+    response = permitted_client.get("/requests/")
+    request_ids = {r.request_id for r in response.context["requests"]}
+    assert request_ids == {"test-request"}
+
+
+def test_requests_index_user_is_output_checker(client_with_user):
+    WorkspaceFactory("test1").create_request("test-request1")
+    WorkspaceFactory("test2").create_request("test-request2")
+    permitted_client = client_with_user(
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply tests/integration/test_views.py:213]: Yeah, I wanted to be explicit in the test setup, so the reader doesn't need to know that.

File: airlock/workspace_api.py [hunk 3]
```diff
@@ -7,6 +7,24 @@
...
+        workspace_names = [d.name for d in settings.REQUEST_DIR.iterdir() if d.is_dir()]
+    else:
+        workspace_names = user.workspaces
+
+    for workspace_name in workspace_names:
+        workspace = Workspace(workspace_name)
+        if not workspace.exists():
+            continue
+
+        releases = [r.name for r in (settings.REQUEST_DIR / workspace_name).iterdir()]
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply airlock/workspace_api.py:43]: Good catch! I've actually fixed this in some wip, but will backport the fix here.

File: tests/unit/test_workspace_api.py [hunk 4]
```diff
@@ -11,7 +11,30 @@
...
+    WorkspaceFactory("test2").create_request("request2")
+
+    assert list(get_releases_for_user(User(1, "test", [], False))) == []
+    assert set(get_releases_for_user(User(1, "test", ["test1"], False))) == {
+        ReleaseRequest(Workspace("test1"), "request1"),
+    }
+    assert set(get_releases_for_user(User(1, "test", [], True))) == {
+        ReleaseRequest(Workspace("test1"), "request1"),
+        ReleaseRequest(Workspace("test2"), "request2"),
+    }
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply tests/unit/test_workspace_api.py:50]: Yes, I prefer not to parameterise small numbers of tests with simple setup. I think the readiblity is much worse when parametrizing, and the isolated failures are not that much of a gain for small parameter sizes.

But can parametrize if you prefer.

File: airlock/workspace_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bloodearnest) [reply airlock/workspace_api.py]: Fixed

File: tests/unit/test_workspace_api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (bloodearnest) [reply tests/unit/test_workspace_api.py]: I've moved to parameterised tests

**Comment by bloodearnest** [comment]
Ok, this is ready for another look.

I've also rebased on main and added fix for User output_checker

File: tests/unit/test_workspace_api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rebkwok) [reply tests/unit/test_workspace_api.py]: I'm not that bothered either way when it's smaller tests - I think I've just got used to reading parametrized tests now

**Review by rebkwok — APPROVED** [review]

**Pull request merged**